lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -3125,6 +3255,36 @@ void PullbackCloner::Implementation::visitSILBasicBlock(SILBasicBlock *bb) {
         break;
       }
       }
+    } else if (auto *tai = dyn_cast<TryApplyInst>(bbArg->getSingleTerminator())) {
+      // try_apply does not provide result, so there is no value to propagate
+      // adjoint to.  Prepare adjoint and associate it "on side", so it will be
+      // used as seed during pullback call emission
+      if (!getPullbackInfo().shouldDifferentiateApplySite(tai))
+        continue;
+
+      LLVM_DEBUG(getADDebugStream() << "Creating adjoint value to active try_apply "
+                 << *tai << " in " << bb->getDebugID() << " destination  ");
+
+      auto *predBB = tai->getParentBlock();
+      switch (getTangentValueCategory(bbArg)) {
+      case SILValueCategory::Object: {
+        auto bbArgAdj = getAdjointValue(bb, bbArg);
+        auto concreteBBArgAdj = materializeAdjointDirect(bbArgAdj, pbLoc);
+        auto concreteBBArgAdjCopy =
+          builder.emitCopyValueOperation(pbLoc, concreteBBArgAdj);
+
+        blockTemporaries[getPullbackBlock(predBB).first].insert(concreteBBArgAdjCopy);
+        auto [_, inserted] = tryApplyAdjoints.try_emplace(tai, concreteBBArgAdjCopy);
+        assert(inserted && "should have unique adjoint for try_apply");
+        break;
+      }
+      case SILValueCategory::Address: {
```

Comment:
With existing tests, we never fall into this case. It's probably worth covering it with tests.

Reply:
Yes, it seems we'd need something like object value with non-loadable tangent.